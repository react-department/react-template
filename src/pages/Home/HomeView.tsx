import { useTranslation } from 'react-i18next';

import Loader from '../../components/Loader/Loader';
import useFetchJokeQuery from '../../store/slices/jokes/apis/jokesRandom';

import type { ReactElement } from 'react';

import styles from './sass/HomeView.module.scss';
/**
 * Home page view
 */
function HomeView(): ReactElement {
  const { data, refetch: onGetJoke, isFetching } = useFetchJokeQuery();
  const { t } = useTranslation();
  return (
    <div>
      <div className={styles.content}>
        {isFetching ? (
          <Loader />
        ) : (
          <>
            <button type="button" onClick={onGetJoke}>{ t('home.getJoke') }</button>
            <p>{data?.value}</p>
          </>
        )}
      </div>
    </div>
  );
}

export default HomeView;
