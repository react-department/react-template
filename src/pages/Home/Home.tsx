import HomeView from './HomeView';

import type { ReactElement } from 'react';

function Home(): ReactElement {
  return (
    <HomeView />
  );
}

export default Home;
