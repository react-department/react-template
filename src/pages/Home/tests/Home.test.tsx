import { render } from '../../../../__tests__/test-utils';
import Home from '../Home';

describe('<Home />', () => {
  test('should render component', async () => {
    const { asFragment } = render(<Home />);
    expect(asFragment()).toMatchSnapshot();
  });
});
