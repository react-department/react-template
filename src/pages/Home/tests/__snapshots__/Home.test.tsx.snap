// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> should render component 1`] = `
<DocumentFragment>
  <div>
    <div
      class="content"
    >
      <div>
        <img
          alt="loader"
          class="appLoader"
          data-testid="loader"
          src="loader.svg"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
